zipkin/transport/scribeclient.py

```diff
@@ -0,0 +1,204 @@
+        cls.host = settings.get(prefix + 'collector')
+        if prefix + 'collector.port' in settings:
+            cls.port = int(settings[prefix + 'collector.port'])
+
+    @classmethod
+    def get_connection(cls):
+        if not cls._client:
+            cls._connection_attempts += 1
+
+            max_retries = CONNECTION_RETRIES[-1]
+            if ((cls._connection_attempts > max_retries) and
+                    not ((cls._connection_attempts % max_retries) == 0)):
+                return
+            if ((cls._connection_attempts < max_retries) and
+                    (cls._connection_attempts not in CONNECTION_RETRIES)):
+                return
+
+            try:
+                cls._client = make_client(
+                    scribe_thrift.Scribe, host=cls.host,
+                    port=cls.port)
+
+                cls._connection_attempts = 0
+            except TTransportException:
+                cls._client = None
+                logger.error("Can't connect to zipkin collector %s:%d"
+                             % (cls.host, cls.port))
+            except Exception:
+                cls._client = None
+                logger.exception("Can't connect to zipkin collector %s:%d"
+                                 % (cls.host, cls.port))
+        return cls._client
+
+    @classmethod
+    def log(cls, trace):
+        if not cls.host:
+            logger.debug('Zipkin tracing is disabled')
+            return
+
+        logger.error('logging trace %s' % trace.trace_id)
+
+        unknown = ('Unknown Exception while logging a trace on '
+                   'zipkin collector %s:%d' % (cls.host, cls.port))
+
+        client = cls.get_connection()
+        if client:
+            try:
+                messages = [base64_thrift_formatter(t, t.annotations)
+                            for t in trace.children()]
+                log_entries = [scribe_thrift.LogEntry('zipkin', message)
+                               for message in messages]
+            except ValueError:
+                logger.exception('Error while serializing trace')
+                return
+
+            try:
+                client.Log(messages=log_entries)
+            except EOFError:
+                cls._client = None
+                logger.error('EOFError while logging a trace on zipkin '
+                             'collector %s:%d' % (cls.host, cls.port))
+            except socket.error as err:
+                cls._client = None
+                if err.errno == errno.EPIPE:
+                    logger.error('Broken pipe while logging a trace '
+                                 'on zipkin collector %s:%d',
+                                 cls.host, cls.port)
+                else:
+                    logger.exception(unknown)
+            except Exception:
+                cls._client = None
+                logger.exception(unknown)
+        else:
+            logger.warn("Can't log zipkin trace, not connected")
```

Comment:
hasn't `logging.warn` been deprecated in python 3? :thinking: 

```suggestion
            logger.warning("Can't log zipkin trace, not connected")
```

Reply:
Don't remember the version but @grigouze have made a PR about this in 2019: #4

Reply:
Actually i've move the file client.py to transport/scribeclient.py without rewriting the content...